src/mito2/src/compaction/twcs.rs

```diff
@@ -195,24 +201,88 @@ impl Picker for TwcsPicker {
     }
 }
 
+#[derive(Default)]
+struct Window {
+    start: Option<Timestamp>,
+    end: Option<Timestamp>,
+    files: Vec<FileHandle>,
+    time_window: i64,
+    overlapping: bool,
+}
```

Comment:
nitpick: We can define a constructor `Window::new(file: FileHandle)` and remove `default()` so files are not empty
```suggestion
struct Window {
    start: Timestamp,
    end: Timestamp,
    files: Vec<FileHandle>,
    time_window: i64,
    overlapping: bool,
}
```

Reply:
fixed in 2b4a00cf0f